jrnl/cli.py

```diff
@@ -478,5 +478,8 @@ def run(manual_args=None):
         journal.write()
 
     elif args.delete:
+        other_entries = [e for e in old_entries if e not in journal.entries]
         journal.prompt_delete_entries()
+        journal.entries += other_entries
```

Comment:
Adding the old entries back at this point means that the journal is still being deleted on disk (by the `write` in `prompt_delete_entries`). This then adds them back. So, the problem here is that the journal is now being written out twice, which is slower, and has the possibility to still delete the journal entries (in case something goes wrong in between the two writes).